### PR TITLE
Enable conditional steps retrievals for evaluators 

### PR DESCRIPTION
**Pull request by reidkwja**
new -> compiled -> bar_lock

### CODE REVIEW

File: tuna/miopen/worker/fin_eval.py [hunk 1]
```diff
@@ -54,10 +54,12 @@ def __init__(self, **kwargs):
   def get_job(self, find_state, set_state, imply_end):
     """Polling to see if job available"""
     self.logger.info('find job: %s', find_state)
+
     if not super().get_job(find_state, set_state, imply_end):
       with self.bar_lock:
         self.num_procs.value -= 1
       return False
```
Comment (cderb) [inline tuna/miopen/worker/fin_eval.py]: This should be evaluated after we've finished checking for all job times. After checking "compiled" and "new". Only when there are no more jobs.

**Review by cderb — CHANGES_REQUESTED** [review]

**Comment by alexandraBara** [comment]
What is the reason for implementing this? the job queue is going away completely once celery is fully implemented but I need to know what the use case is here, in case this is something i need to consider. Keep in mind that the entire job queue is being moved to celery -  which handles the implementation details for us.

**Comment by cderb** [comment]
@alexandraBara We need the evaluation process to be able to retrieve jobs from the new state as well. We are also trying to prioritize compiled jobs.

File: tuna/miopen/worker/fin_eval.py [hunk 2]
```diff
@@ -268,6 +259,25 @@ def close_job(self):
     self.set_job_state('evaluated')
     self.clean_cache_table()
 
+  def get_job(self, find_state, set_state, imply_end):
+    """Polling to see if job available"""
+    self.logger.info('find job: %s', find_state)
+
+    if not super().get_job(find_state, set_state, imply_end):
+      return False
+    return True
```
Comment (cderb) [inline tuna/miopen/worker/fin_eval.py:268]: Safe to remove this function.

**Review by cderb — APPROVED** [review]

**Review by alexandraBara — APPROVED** [review]

File: tuna/miopen/worker/fin_eval.py (hunk 2 above)
Comment (msaudulhassan) [inline tuna/miopen/worker/fin_eval.py:268]: This can be replaced with
```python
return super().get_job(find_state, set_state, imply_end)
```

**Review by msaudulhassan — APPROVED** [review]
LGTM. Here's a minor suggestion though

**Pull request merged**